### DISCUSSION TITLE
Crash during load story mode

### DISCUSSION THREAD

**Discussion by ArsenVirus** [scripthookvdotnet/scripthookvdotnet#1417] [Q&A]
### What modules are you seeing the problem on?

ScriptHookVDotNet.asi (runtime)

### SHVDN Version

3.6.0

### Game Version

 v1.0.3095

### Bug Description

[19:09:06] [DEBUG] Loading API from .\ScriptHookVDotNet2.dll ...
[19:09:06] [DEBUG] Loading API from .\ScriptHookVDotNet3.dll ...
[19:09:06] [ERROR] Failed to create console: System.NullReferenceException: Object reference not set to an instance of an object
  at System.AppDomain.CreateInstanceFromAndUnwrap (System.String assemblyName, System.String typeName) [0x00000] in <de4d51d88e3b4eea89a8d6a3a9854818>:0 
  at <Module>.?A0xc1aa4bd5.ScriptHookVDotnet_ManagedInit () [0x00188] in <0f1133baa2e243d2802541074a8dc0b4>:0 
  _**I use steam deck**_

### Steps To Reproduce

1. Download ScriptHookVdotNet from the official site.
2. Paste the files into the root folder of the game.
3. Start the game!

### The Current Behavior

after starting the story mode, the game crashes

### The Expected Behavior

Game crash

**Comment by kagikn** [comment]
Another error report by a user who use Linux and that may not be responsible to SHVDN's codebase but installation issues. Similar issues are already reported with `AppDomain.CreateInstanceFromAndUnwrap` throwing `NullReferenceException` where all the users reported such issues are running in Linux environments.

I guess you need to install Microsoft's .NET Framework 4.8? Sorry, I don't have a Linux description environment yet, so you have to find out a way yourself.

**Comment by ArsenVirus** [comment]
Thanks for the answer. The problem is that I already downloaded the .NET Framework 4.8 from Microsoft and I found this - https://github.com/scripthookvdotnet/scripthookvdotnet/issues/1093 but the user deleted the answer. Is there any other way to fix this error. _**Sorry for the English translation, I use Google Translate**_

**Comment by kagikn** [comment]
Could you please post a solution if you find one, so someone else can find out how to deal with the same or similar issues easily? Sorry, but I'm kind of getting tired of similar issues. Almost the same as how https://github.com/scripthookvdotnet/scripthookvdotnet/issues/1318#issuecomment-1741514296 gets closed I guess?

**Comment by ArsenVirus** [comment]
I have a question. What should I do about it? env WINEPREFIX="~/.steam/debian-installation/steamapps/compatdata/271590/pfx" winetricks --force d3dcompiler_42 d3dcompiler_43 d3dcompiler_47 d3dx10 d3dx10_43 d3dx11_42 d3dx11_43 d3dx9 dotnet452 dotnet48 v crun2019
Sorry for the question, just new to this. thank you

**Comment by kagikn** [comment]
Time to convert to a discussion I guess. I'm preparing for a new Linux desktop environment because I'm getting tired of seeing similar questions and I'm feeling it's the only way to find out a solid solution by myself, but I haven't finished doing and have no idea how to deal with the issue using winetricks.

**Comment by ZeusOfTheCrows** [comment]
i managed to get it working with a simple test script ([Mini-Map Disabled On Foot](https://www.gta5-mods.com/scripts/mini-map-disabled-on-foot-ty6377)):

- using 
	- proton-ge 8.27
	- winetricks 20230212-next with wine-8.20 (Staging)
	- launch options `WINEDLLOVERRIDES="dinput8,d3dcompiler_46=n,b" %command%` (probably not relevant)

- use winetricks to install .net 4.8:
	- installation fails with protontricks
	- .net 4.0 doesn't work, neither does 4.6.2. other versions i have not tested
	- `WINEPREFIX="/home/zeus/Games/steamapps/compatdata/271590/pfx" winetricks dotnet48` would probably work, but i did the following:
		- `WINEPREFIX="/home/zeus/Games/steamapps/compatdata/271590/pfx" winetricks`
		- ignore all warnings
		- select default wineprefix → install a windows dll or component → dotnet 4.8
		- install .net using gui wizards (bear in mind there are two)
- use protontricks to run winecfg and set the windows version back to 10
	- (otherwise it will default to xp, and gtav will refuse to launch)
- upon launching gtav, an error will pop-up saying it failed to launch and proffering more info, click `no` and it should launch normally

installed on a fairly heavily modded game, but i'm pretty sure these are the only relevant steps
